docs/Test.md

```diff
@@ -0,0 +1,143 @@
+# Test
+
+Tests serve many different purposes. They can:
+
+- Validate the requirements/expectations of stakeholders are met
+- That an end user can achieve their goal
+- Act as documenatation/reference material
+- Confirm integration works as expected between areas of function
+- Validate we do not regress functionality or behaviour we care about
+
+We want to enable and achieve all of these goals in the UI testing done in this
+repo. To this end, we require, depending on the functionality being added or
+changed, different types of tests. The types of test expected, and the
+rationale for these are detailed below.
+
+## Style of test
+
+We have two approaches of testing at the unit level - `Behavioural driven development`
+and standard `Test driven development`.
+
+`Behavioural driven development` tests focus on exactly that - behaviour. They
+do not target individual functons, but instead test an end to end behaviour
+(and all the functions which contribute to it), and validate that for a given
+set of actions or inputs, that the expected output/end goal is met. These
+behaviours should focus on the user of the component. Depending on the type of
+code being developed, the user may be another developer (using the component
+for example) or the end user of the UI.
+
+`Test driven development` tests conversely target individual functions in a
+more traditional unit test style - validating that for a given input, the
+expected output is returned.
+
+Both styles of test should live in the same directory as the components they
+are testing.
+
+Depending on the code being covered, the tools used to enable these test styles
+may be different, as detailed below.
+
+### Test Driven Development
+
+Code in the following directories are expected to be driven by standard test
+driven development. They use _Jest_ as a test runner, using standard Jest
+test assets, such as `describe`, `it` and `expect`:
+
+- `client/Hooks`
+- `client/Contexts`
+
+### Behavioural Driven Development
+
+Code in the following directories are expected to be driven by behavioural
+driven deveopment. They use _Jest_ as a test runner, using standard Jest test
+assets, such as `describe`, `it` and `expect`, and _RTL_ to mount and interact
+with React component code:
+
+- `client/Elements`
+- `client/Groups`
+- `client/Panels`
+
+### Server Testing
+
+The strimzi-ui server should be tested from outside the server - like how a
```

Comment:
`The strimzi-ui server should be tested from outside the server - like how a user would use it`. I think this might need a slight grammar tweak :)  Honestly, I am not sure if this sentence adds much. The main thing to know is the server code is/should be tested in a behavioural manner, as a user would acessing it (indirectly) via the UI. I think the second sentence (with a few extra words) could do the job.